### PR TITLE
Better sorting and filtering for bounties

### PR DESCRIPTION
**Pull request by manuelurenah**

### CODE REVIEW

File: src/components/Bounty/BountyEntryUpsertForm.tsx [hunk 1]
```diff
@@ -42,17 +41,28 @@ import { getBountyCurrency, getMainBountyAmount } from '~/components/Bounty/boun
 import { BountyEntryMode, BountyType, Currency } from '@prisma/client';
 import { CurrencyIcon } from '~/components/Currency/CurrencyIcon';
 import { formatKBytes } from '~/utils/number-helpers';
-import { AlertWithIcon } from '~/components/AlertWithIcon/AlertWithIcon';
+// import { AlertWithIcon } from '~/components/AlertWithIcon/AlertWithIcon';
```
Comment (lrojas94) [inline src/components/Bounty/BountyEntryUpsertForm.tsx:44]: Remove if unused

File: src/components/Bounty/BountyEntryUpsertForm.tsx [hunk 2]
```diff
@@ -375,22 +360,25 @@ export function BountyEntryUpsertForm({ bountyEntry, bounty }: Props) {
...
             dropzoneProps={{
               maxFiles: 10,
-              accept: [...IMAGE_MIME_TYPE, ...ZIP_MIME_TYPE],
+              accept: acceptedFileTypes,
             }}
-            renderItem={(file, onRemove) => {
+            renderItem={(file, onRemove, onUpdate) => {
+              const metadata = (file.metadata ?? {}) as BountyEntryFileMeta;
+              // const currency = metadata.currency || getBountyCurrency(bounty);
+              // const unlockAmount = metadata.unlockAmount || 0;
```
Comment (lrojas94) [inline src/components/Bounty/BountyEntryUpsertForm.tsx:380]: Same

File: src/server/services/bounty.service.ts [hunk 3]
```diff
@@ -66,7 +67,15 @@ export const getAllBounties = <TSelect extends Prisma.BountySelect>({
     });
   }
 
-  const orderBy: Prisma.BountyFindManyArgs['orderBy'] = [{ complete: 'asc' }];
+  if (status) {
+    if (status === BountyStatus.Open) AND.push({ complete: false, expiresAt: { gt: new Date() } });
+    else if (status === BountyStatus.Awarded) AND.push({ complete: true, entries: { some: {} } });
```
Comment (lrojas94) [inline src/server/services/bounty.service.ts:72]: This is interesting, `some: {}` works? I'd assume you need to provide something here 🤔 The more you know 👍

**Review by lrojas94 — APPROVED** [review]

File: src/components/Bounty/BountyEntryUpsertForm.tsx (hunk 2 above)
Comment (manuelurenah) [reply src/components/Bounty/BountyEntryUpsertForm.tsx:380]: We are keeping this in case we are using it when we allow open entries

**Pull request merged**